rust/experimental/query_engine/engine-columnar/src/pipeline/filter.rs

```diff
@@ -402,17 +407,164 @@ impl TryFrom<&LogicalExpression> for Composite<FilterPlan> {
+            LogicalExpression::Scalar(_) => Err(Error::NotYetSupportedError {
                 message: format!("Logical expression not yet supported {logical_expr:?}"),
             }),
         }
     }
 }
 
+impl TryFrom<&ContainsLogicalExpression> for FilterPlan {
+    type Error = Error;
+
+    fn try_from(contains_expr: &ContainsLogicalExpression) -> Result<Self> {
+        let left_arg = BinaryArg::try_from(contains_expr.get_haystack())?;
+        let right_arg = BinaryArg::try_from(contains_expr.get_needle())?;
+
+        match left_arg {
+            BinaryArg::Column(left_column) => {
+                let mut attrs = None;
+                let left_expr = match left_column {
+                    ColumnAccessor::ColumnName(left_col_name) => col(left_col_name),
+                    ColumnAccessor::StructCol(left_struct_name, left_struct_field) => {
+                        col(left_struct_name).field(left_struct_field)
+                    }
+                    ColumnAccessor::Attributes(attrs_identifier, attrs_key) => {
+                        attrs = Some((attrs_identifier, attrs_key));
+                        // for now we assume that text contains is always applied to the str column
+                        col(consts::ATTRIBUTE_STR)
+                    }
+                };
+
+                let right_expr =
+                    match right_arg {
+                        BinaryArg::Literal(right_lit) => try_static_scalar_to_literal(&right_lit)?,
+                        _ => return Err(Error::NotYetSupportedError {
+                            message:
+                                "text contains predicate comparing column left to non literal right"
+                                    .into(),
+                        }),
+                    };
+
+                let contains_expr = contains(left_expr, right_expr);
+                Ok(match attrs {
+                    None => FilterPlan::from(contains_expr),
+                    Some((attrs_identifier, attrs_key)) => {
+                        FilterPlan::from(AttributesFilterPlan::new(
+                            col(consts::ATTRIBUTE_KEY)
+                                .eq(lit(attrs_key))
+                                .and(contains_expr),
+                            attrs_identifier,
+                        ))
+                    }
+                })
+            }
+            BinaryArg::Literal(left_lit) => {
+                let left_expr = try_static_scalar_to_literal(&left_lit)?;
+
+                let mut attrs = None;
+                let right_expr = match right_arg {
+                    BinaryArg::Column(right_column) => match right_column {
+                        ColumnAccessor::ColumnName(right_col_name) => col(right_col_name),
+                        ColumnAccessor::StructCol(right_struct_name, right_struct_field) => {
+                            col(right_struct_name).field(right_struct_field)
+                        }
```

Comment:
I think this repeats the handling for left_column above, it might help to add a new function.

Reply:
Good calll @jmacd , made this change